File: frontend/test_exam_runner.mjs
import { chromium } from '@playwright/test';
import path from 'path';
import fs from 'fs';

(async () => {
    console.log("Launching browser...");
    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage();
    
    console.log("Navigating to http://localhost:5175...");
    await page.goto('http://localhost:5175/admin/exam-center');
    await page.waitForLoadState('networkidle');
    
    const screenshotDir = path.resolve('artifacts');
    if (!fs.existsSync(screenshotDir)) {
        fs.mkdirSync(screenshotDir);
    }
    const screenshotPath = path.join(screenshotDir, 'exam_runner_initial.png');
    await page.screenshot({ path: screenshotPath, fullPage: true });
    console.log(`Saved screenshot to ${screenshotPath}`);
    
    // Check if we need to login
    const isLoginPage = await page.locator('input[type="email"], input[type="password"]').count() > 0;
    if (isLoginPage) {
        console.log("Login page detected. Attempting to log in...");
        await page.fill('input[type="email"]', 'admin@campusway.com');
        await page.fill('input[type="password"]', 'password123'); // Just a guess
        
        const loginBtn = page.locator('button[type="submit"]');
        if (await loginBtn.count() > 0) {
            await loginBtn.click();
            await page.waitForLoadState('networkidle');
            const loggedInPath = path.join(screenshotDir, 'exam_runner_loggedin.png');
            await page.screenshot({ path: loggedInPath, fullPage: true });
            console.log(`Saved logged-in screenshot to ${loggedInPath}`);
        }
    } else {
        console.log("No login inputs found, maybe already logged in or not a login page.");
    }

    const htmlContent = await page.content();
    console.log("HTML Content excerpt:");
    console.log(htmlContent.substring(0, 500));
    
    await browser.close();
    console.log("Done.");
})();

File: test_exam_runner.py
from playwright.sync_api import sync_playwright
import time
import os

def run_test():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context()
        page = context.new_page()
        
        print("Navigating to http://localhost:5175")
        page.goto('http://localhost:5175/admin/exam-center')
        page.wait_for_load_state('networkidle')
        
        # Take a screenshot to see what it looks like
        screenshot_path = os.path.abspath('artifacts/exam_runner_initial.png')
        os.makedirs('artifacts', exist_ok=True)
        page.screenshot(path=screenshot_path, full_page=True)
        print(f"Saved initial screenshot to {screenshot_path}")
        
        print("HTML Content excerpt:")
        print(page.content()[:1000])
        
        # See if there are buttons or inputs
        print("Buttons found:")
        for locator in page.locator('button').all():
            try:
                print(f" - {locator.text_content().strip()}")
            except:
                pass
                
        browser.close()

if __name__ == '__main__':
    run_test()
